src/main/java/com/enderio/base/common/handler/GrindingHandler.java

```diff
@@ -3,53 +3,57 @@
 import com.enderio.EnderIO;
 import com.enderio.base.common.init.EIOItems;
 import net.minecraft.util.RandomSource;
+import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
-import net.minecraftforge.event.GrindstoneEvent;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraftforge.event.entity.player.PlayerInteractEvent;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 import net.minecraftforge.fml.common.Mod;
 
 @Mod.EventBusSubscriber(modid = EnderIO.MODID)
 public class GrindingHandler {
     @SubscribeEvent
-    public static void onGrindstonePlace(GrindstoneEvent.OnPlaceItem event) {
-        if ((event.getTopItem().is(Items.DEEPSLATE) || event.getTopItem().is(Items.COBBLED_DEEPSLATE)) && event.getBottomItem().is(Items.FLINT)) {
-            event.setOutput(new ItemStack(EIOItems.GRAINS_OF_INFINITY.get(), 1));
-            event.setXp(0);
-        } else if (event.getTopItem().is(Items.COAL) && event.getTopItem().getCount() >= 3 && event.getBottomItem().is(Items.FLINT)) {
-            event.setOutput(new ItemStack(EIOItems.POWDERED_COAL.get(), 1));
-            event.setXp(0);
+    public static void onRightClickBlock(PlayerInteractEvent.RightClickBlock event) {
+        Player player = event.getEntity();
+        Block target = event.getLevel().getBlockState(event.getPos()).getBlock();
+        ItemStack mainhand = player.getMainHandItem();
+        ItemStack offhand = player.getOffhandItem();
+        if (!(player.isCrouching())
+            || (target != Blocks.OBSIDIAN && target != Blocks.GRINDSTONE && target != Blocks.CRYING_OBSIDIAN)
+            || !(offhand.is(Items.FLINT))) {
+            return;
```

Comment:
Split this guard clause into two
```suggestion
        if (!player.isCrouching())
            return;
        if (target != Blocks.OBSIDIAN && target != Blocks.GRINDSTONE && target != Blocks.CRYING_OBSIDIAN) || !(offhand.is(Items.FLINT)))
            return;
```